### DISCUSSION TITLE
TypeScript config and specific error

### DISCUSSION THREAD

**Discussion by ghost** [LinkedSoftwareDependencies/Components.js#104] [Q&A]
Hello all,

I'm working with componentsjs and attempting to use the typescript bundling and componentsjs-generator builds, and I keep running into this error: Error: Failed to get module element SomeRandomComponent from module my-module-name. I'm asking to understand where in the instantiation process this error is built from? Is it an error parsing the module file, a config file, or does it mean it is missing a pointer? When I instantiate, I make sure the component instantiation is pointing to the correct @id key for the component instance, so at this point I know I am misunderstanding about the registration and instantiation process..

**Comment by rubensworks** [comment]
This usually means that either you made a typo in your config file (the component has a different name), or the component is not being exported by your package. In both cases, you can check the exact name of your component in generated your `components/` directory.

If both of these are not the problem, then it could also be related to component discovery, where your mainModulePath may not be set correctly, and Components.js doesn't find your components.
